src/cmdlib.sh

```diff
@@ -628,6 +628,41 @@ strip_out_lockfile_digests() {
     mv "$1.tmp" "$1"
 }
 
+# Strip out the arch field from lockfiles to make them archless.
+strip_out_lockfile_arches() {
+    python3 -c '
+import sys, json
+lockfile = json.load(sys.stdin)
+new_packages = {}
+for pkg, meta in lockfile["packages"].items():
+    evra = meta.get("evra")
+    if evra is None or evra.endswith(".noarch"):
+        new_packages[pkg] = meta
+        continue
+    new_packages[pkg] = {
+        "evr": evra[:evra.rindex(".")]
+    }
+lockfile["packages"] = new_packages
+json.dump(lockfile, sys.stdout)
+' < "$1" > "$1.tmp"
+    mv "$1.tmp" "$1"
+}
+
+# Create the autolock for a given version
+generate_autolock() {
+    local version=$1; shift
```

Comment:
Is the `shift` necessary here since we don't use positional vars any more in this function?

Reply:
Not necessary, but I think it's a good convention. :)

Reply:
yes, a good convention but only if it has an impact. otherwise it's confusing, at least to me.

Reply:
Conceptually, I see it as "consuming" the argument. This e.g. prevents code from using `$1`. It's also cleaner if we add more arguments down the line to this function.

Reply:
I differ here slightly, but definitely not worth discussing it further :)